server/sonar-web/src/main/js/app/components/ProjectAdminContainer.tsx

```diff
@@ -17,20 +17,22 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-import React from 'react';
+import * as React from 'react';
 import handleRequiredAuthorization from '../utils/handleRequiredAuthorization';
+import { Component, Branch } from '../types';
 
-export default class ProjectAdminContainer extends React.PureComponent {
-  /*::
-  props: {
-    component: {
-      configuration?: {
-        showSettings: boolean
-      }
-    }
-  };
-  */
+interface Props {
+  children: JSX.Element;
+  branch?: Branch;
+  branches: Branch[];
```

Comment:
oh... I anticipate some git conflicts here... 😥

Reply:
Should I drop my rewrite of this component ?

Reply:
Oh no, not at all! It was just a comment without any appeal.